### PR TITLE
Feat: run harness parallel tests in parallel

### LINKED ISSUE #657: Run harness tests in parallel
It takes about 10 minutes to run all the terraform provider tests 
image

Can we alter them to run them in parallel to each other?

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
closes #657 

### Solution

1. reverted the revert.
2. updated the integration tests to avoid race conditions.

### CODE REVIEW

File: env0/data_template.go [hunk 1]
```diff
@@ -27,6 +27,11 @@ func dataTemplate() *schema.Resource {
 				Optional:     true,
 				ExactlyOneOf: []string{"name", "id"},
 			},
+			"description": {
```
Comment (TomerHeber) [inline env0/data_template.go:30]: required for the integration testing. No harm is adding it.

File: tests/harness.go [hunk 2]
```diff
@@ -24,16 +25,27 @@ func main() {
 	log.Println(len(testNames), " tests to run")
 	buildFakeTerraformRegistry()
 	destroyMode := os.Getenv("DESTROY_MODE")
+
+	var wg sync.WaitGroup
```
Comment (TomerHeber) [inline tests/harness.go:29]: the changes here were already reviewed - this was a revert.

File: tests/integration/022_templates/expected_outputs.json [hunk 3]
```diff
@@ -1 +1,4 @@
-{}
+{
```
Comment (TomerHeber) [inline tests/integration/022_templates/expected_outputs.json:1]: added output test.

**Comment by GiliFaroEnv0** [comment]
[Add State Encryption As an Optional Value In Values.yaml](https://trello.com/c/sniNbaCx/10194-add-state-encryption-as-an-optional-value-in-valuesyaml)

**Review by GiliFaroEnv0 — APPROVED** [review]
NICE!
great way

**Pull request merged**